dotnet/src/SemanticKernel.Abstractions/AI/Embeddings/ITextEmbeddingService.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.SemanticKernel.AI.Embeddings;
+
+/// <summary>
+/// Represents a generator of text embeddings of type <c>float</c>.
+/// </summary>
+public interface ITextEmbeddingService : IEmbeddingGenerationService<string, float>
```

Comment:
this is missing "Generation". Overall I really dislike the long names and having the "Service" suffix everywhere, it makes the API verbose and pedantic. If we touch this again we should use the opportunity to simplify.
Using ServiceCollection it would look something like `Kernel.Services.Add<ITextCompletion, ...>(...)`

Reply:
looking at some examples:
* ILogger vs ILoggingService
* ITextRendering vs ITextRenderingService
* IOpenApiDocumentParser vs IOpenApiDocumentParserService

we could use
* `ITextEmbeddingGenerator` instead of `ITextEmbeddingGenerationService`
* `ITextGenerator` instead of `ITextCompletion` / `ITextCompletionService`
* `IChatMessageGenerator` instead `IChatCompletion` / `IChatCompletionService`
* etc.

Reply:
My 5 cents:

I think the Service suffix does a very good job of denoting that the interfaces/classes having it are related to the infrastructure layer and has nothing to do with domain layer - Kernel. That, BTW, allow quickly identify and prevent cases of domain logic spilling into the services during development or on PR level.

So, the verbosity is not necessarily a bad thing as long at service a dedicated purpose and has a meaning.

Reply:
I'd be happy with *Generator suffix, as it fits the pattern nicely.

Rationale:
ILogg**er** is a service that generates Logs
IOpenApiDocumentPars**er** is a service that parses documents
A TextCompletion is the product of a service -- not the producer. It's literally the object you get back. If I sit down and start intellisensing with an ITextCompletion object, I'm certainly not going to expect it to be a conduit for hitting a service... I'd expect it to have string value, tokens used, etc.

Reply:
ITextGenerator isn't sufficiently descriptive or accurate either. ITextCompletionGenerator is fine, but is longer than *Service.

If we don't like *Service, we could use:

ITextEmbeddingGenerator
IImageGenerator 
ITextCompletionGenerator or ITextCompletionProvider
IChatCompletionGenerator or IChatCompletionProvider

Reply:
Reverted. We can cover services naming in another PR.